apps/crn-server/src/data-providers/contentful/teams.data-provider.ts

```diff
@@ -0,0 +1,241 @@
+
+    const team = await environment.getEntry(id);
+
+    const teamUpdated = await team.patch([
+      {
+        op: 'replace',
+        path: '/fields/tools/en-US',
+        value: publishedTools.map((tool) => ({
+          sys: {
+            type: 'Link',
+            linkType: 'Entry',
+            id: tool.sys.id,
+          },
+        })),
+      },
+    ]);
+
+    await teamUpdated.publish();
+  }
+
+  async create(input: TeamCreateDataObject): Promise<string> {
+    const environment = await this.getRestClient();
+
+    const { tools, ...plainInput } = input;
+    let toolsInput = {};
+
+    if (tools) {
+      const cleanTools = getCleanTools(tools);
+      const publishedTools = await createAndPublishTools(
+        environment,
+        cleanTools,
+      );
+      toolsInput = {
+        tools: {
+          'en-US': publishedTools.map((tool) => ({
+            sys: {
+              type: 'Link',
+              linkType: 'Entry',
+              id: tool.sys.id,
+            },
+          })),
+        },
+      };
+    }
+
+    const newEntry = await environment.createEntry('teams', {
+      fields: {
+        ...addLocaleToFields(plainInput),
+        ...toolsInput,
+      },
+    });
+
+    return newEntry.sys.id;
+  }
+}
+
+export const teamUnreadyResponse = {
+  labCount: 0,
+  members: [],
+  pointOfContact: undefined,
+  proposalURL: undefined,
+};
+
+export const parseContentfulGraphQlTeams = (item: TeamItem): TeamDataObject => {
+  const expertiseAndResourceTags = (item.expertiseAndResourceTags || []).reduce(
+    (tags: string[], tag) => {
+      if (tag) {
+        tags.push(tag);
+      }
+      return tags;
+    },
+    [],
+  );
+
+  const tools = (item.toolsCollection?.items || []).reduce(
+    (teamTools: TeamTool[], tool) => {
+      if (!tool || !tool.name || !tool.url) {
+        return teamTools;
+      }
+
+      const { name, url, description } = tool;
+
+      return [
+        ...teamTools,
+        {
+          name,
+          url,
+          description: description ?? undefined,
+        },
+      ];
+    },
+    [],
+  );
+
+  return {
+    id: item.sys.id ?? '',
+    displayName: item.displayName ?? '',
+    inactiveSince: item.inactiveSince ?? undefined,
+    projectTitle: item.projectTitle ?? '',
+    lastModifiedDate: new Date(item.sys.publishedAt).toISOString(),
+    expertiseAndResourceTags,
+    tools,
+    projectSummary: item.projectSummary ?? undefined,
+
+    // TODO implement below when users (CRN-1164),
+    // labs (CRN-1263) and RO (CRN-1253) are ready
+    ...teamUnreadyResponse,
+  };
+};
+
+const getCleanTools = (tools: TeamTool[]) =>
+  tools.map((tool) =>
+    Object.entries(tool).reduce(
+      (acc, [key, value]) =>
+        value?.trim && value?.trim() === ''
+          ? acc // deleted field
+          : { ...acc, [key]: value },
+      {} as TeamTool,
+    ),
+  );
+
+const createAndPublishTools = (environment: Environment, tools: TeamTool[]) =>
+  Promise.all(
```

Comment:
if my suggestion https://github.com/yldio/asap-hub/pull/2759/files#r1138461381 is implemented we would want to use  Promise.allSettled here.